Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -706,15 +727,6 @@ namespace Babylon
     void NativeEngine::Dispose()
     {
         m_cancellationSource->cancel();
-
-        // TODO: clean up bound vertex array
-
-        // This collection contains bgfx data, so it must be cleared before bgfx::shutdown is called.
-        for (const auto handle : m_programDataCollection)
-        {
-            ProgramData::Delete(handle);
-        }
```

Comment:
What is the equivalent new thing that synchronously deletes `ProgramData` instances when the `NativeEngine` is disposed?

Reply:
This change should make `ProgramData` follow the same pattern as every other resource type: formally owned by JavaScript variables, typically disposed manually through `NativeEngine`, and destructed by JavaScript if it hasn't been properly disposed by shutdown.